go/apps/api/routes/v2_apis_list_keys/handler.go

```diff
@@ -89,59 +88,56 @@ func (h *Handler) Handle(ctx context.Context, s *zen.Session) error {
 		return err
 	}
 
-	// 4. Get API from database
-	api, err := db.Query.FindApiByID(ctx, h.DB.RO(), req.ApiId)
-	if err != nil {
-		if db.IsNotFound(err) {
-			return fault.New("api not found",
-				fault.Code(codes.Data.Api.NotFound.URN()),
-				fault.Internal("api not found"), fault.Public("The requested API does not exist or has been deleted."),
-			)
-		}
-		return fault.Wrap(err,
-			fault.Code(codes.App.Internal.ServiceUnavailable.URN()),
-			fault.Internal("database error"), fault.Public("Failed to retrieve API information."),
-		)
-	}
```

Comment:
I actually think we should keep this query, because it is easily cachacble and having a cache for the api's wouldn't be bad because we will need it in multiple places aswell. 

Basically in each api endpoint we can re-use it.

Without this query, we would falsely return 200 OK and an empty keys array instead of a 404 error. 

And then there is that we are selecting the same information very often which is unnecessary overhead espeically when selecting alot of keys.
With that we can also do the perms check e.g decryption without having to query keys first.

Reply:
Sure, mind sharing an example for caching?

Reply:
Caches are created in go/internal/services/caches/caches.go
Passed down in go/apps/api/routes/register.go
Used in go/internal/services/keys/get.go

Reply:
@Flo4604 Imma leave this to a separate I'm a bit lost I gotta admit.